hwy/tests/convert_test.cc

```diff
@@ -176,6 +176,168 @@ HWY_NOINLINE void TestAllBitCast() {
 #endif
 }
 
+template <class TTo>
+struct TestResizeBitCastToOneLaneVect {
+  template <typename T, class D>
+  HWY_NOINLINE void operator()(T /*unused*/, D d) {
+    const size_t N = Lanes(d);
+    if (N == 0) {
+      return;
+    }
+
+    auto from_lanes = AllocateAligned<T>(N);
+    HWY_ASSERT(from_lanes);
+    auto v = Iota(d, 1);
+    Store(v, d, from_lanes.get());
+
+    const size_t num_of_bytes_to_copy = HWY_MIN(N * sizeof(T), sizeof(TTo));
+
+    int8_t active_bits_mask_i8_arr[sizeof(TTo)] = {};
+    for (size_t i = 0; i < num_of_bytes_to_copy; i++) {
+      active_bits_mask_i8_arr[i] = int8_t{-1};
+    }
+
+    TTo active_bits_int_mask;
+    CopyBytes<sizeof(TTo)>(active_bits_mask_i8_arr, &active_bits_int_mask);
+
+    const FixedTag<TTo, 1> d_to;
+    TTo expected_bits = 0;
+    memcpy(&expected_bits, from_lanes.get(), num_of_bytes_to_copy);
+
+    const auto expected = Set(d_to, expected_bits);
+    const auto v_active_bits_mask = Set(d_to, active_bits_int_mask);
+
+    const auto actual_1 = And(v_active_bits_mask, ResizeBitCast(d_to, v));
+#if HWY_HAVE_SCALABLE || HWY_TARGET == HWY_SVE_256 || HWY_TARGET == HWY_SVE2_128
```

Comment:
Can we avoid this (and the previous) `#if` somehow?

Reply:
The reason for the #if statement here is that the `ZeroExtendResizeBitCast(d_to, v)` overload treats `v` as a full vector that has `Lanes(DFromV<decltype(v)>())` lanes on RVV/SVE targets and as it is possible for `Lanes(d_from)` to be less than `Lanes(DFromV<decltype(v)>())` on RVV/SVE targets.

The `ZeroExtendResizeBitCast(d_to, d_from, v)` overload always treats `v` as a vector that has `Lanes(d_from)` lanes, even in the case where `Lanes(DFromV<decltype(v)>()) > Lanes(d_from)` is true (which is possible on RVV/SVE).

Reply:
I see, thanks. We have to mask out the other lanes on SVE/RVV, but not other platforms.

Then I wonder: for user clarity, and avoiding surprising changes in behaviour across platforms, should we only provide the form with 3 arguments d_to, d_from, v_from?
The downside is only providing an extra argument, right? But this often happens, for example Reverse2() has a d-arg which is mostly unused on x86 or fixed-size vector, but the op still requires that argument for consistency across platforms.

Reply:
> I see, thanks. We have to mask out the other lanes on SVE/RVV, but not other platforms.
> 
> Then I wonder: for user clarity, and avoiding surprising changes in behaviour across platforms, should we only provide the form with 3 arguments d_to, d_from, v_from? The downside is only providing an extra argument, right? But this often happens, for example Reverse2() has a d-arg which is mostly unused on x86 or fixed-size vector, but the op still requires that argument for consistency across platforms.

I have removed the 2-argument version of the ZeroExtendResizeBitCast function and only kept the 3-argument version of the ZeroExtendResizeBitCast function for user clarity to avoid surprising changes in behavior across platforms.